shared/src/utils/vault.ts

```diff
@@ -62,15 +61,16 @@ export const getUserDepositQueueNodes = (
 
 export const getUserDepositQueueAmount = async (
   vault: vaultTypes.Vault,
-  connection: anchor.web3.Connection,
   userKey: PublicKey
 ): Promise<BigNumber> => {
-  const depositQueueNodes = getUserDepositQueueNodes(vault, userKey);
-  const totalDeposits = await Promise.all(
-    depositQueueNodes.map((node) => connection.getBalance(node))
+  const depositNodes = vault.depositQueue.filter((node) =>
+    node.info.userKey.equals(userKey)
   );
 
-  return BigNumber.from(totalDeposits.reduce((val, acc) => (acc += val), 0));
+  return depositNodes.reduce(
+    (acc, node) => acc.add(BigNumber.from(node.info.amount.toString())),
+    BigNumber.from(0)
+  );
 };
 
 export const getUserWithdrawQueueNodes = (
```

Comment:
You can completely remove the `getUserWithdrawQueueNodes` and `getUserDepositQueueNodes` functions